doc/iterations/2023.10/shapes/color_management_in_view.adoc

```diff
@@ -0,0 +1,106 @@
+= (M) Color Management in View
+
+== Problems
+
+It is possible to set `None` for color properties of a Node/Edge Description.
+Depending on the case, `None` means `black`, `white` or `transparent`.
+
+It is possible to set `None` for color properties of a Form Widget Description.
+Depending on the case, `None` means `theme.primary`, `theme.secondary` or `transparent`.
+
+Some color properties of a Node/Edge/Widget Description are mandatory, which means when `None` is selected, an error message is displayed.
+That's ok but we should review which color property is really mandatory or not.
+
+Furthermore, when a View is created non Color Palette is provided by default.
+The result for color properties of a Node/Edge/Widget Description is that there is no other options than `None`. 
+
+== Key results
+
+* Specifier always know which color will be displayed at the end, corresponding exactly to what is set for each color property.
+* A set of predefined colors is accessible in color properties.
+
+== Solutions
+
+1) Preferred solution
+
+`None` doesn't exist anymore as a possible value for color properties.
+
+All color properties are mandatory.
+
+Theme colors are provided by default (need to choose between):
+
+* A) the theme colors are accessible in the Select widget of color properties by default.
+No need to declare them in a Color Palette, they are accessible everywhere.
+* B) the theme colors are accessible in the Select widget of color properties by default, and they come from a color palette created along with the View and this color palette cannot be modified/deleted.
```

Comment:
I would go for the second option since it would be way easier. Otherwise, we would have to create some additional model with default values and then handle it as a special use case at tons of different locations. For example, we wouldn't want it to be visible or modifiable in the explorer, the details view, or dialogs, etc. This would create tons of special use cases to handle today even if I believe that some default library will one day be the best way to provide several key user experience improvements.

I would move to this direction in two steps.

- We would also need some special colors like "transparent", "inherit", "none" and use some of them as default values when we are creating new elements.
- We should probably provide every single one of the raw MUI colors like [here](https://mui.com/material-ui/customization/color/#color-palette) in various palette (red, pink, blue, etc) with their proper name
- We may also need some [special colors](https://mui.com/material-ui/customization/palette/) like `palette.primary.light`, `palette.secondary.dark` etc to create theme independent representations (for example a for with buttons matching the `palette.error.main` color to perform destructive actions)

This way, we would have a couple of default palette with special CSS Colors (transparent, none, inherit), raw MUI colors (blue 400, grey 800), special theme colors (palette.primary.light). We would need some special treatment for those special theme colors in order to interpret them as JS values instead of CSS colors but that's quite easy.

Reply:
+1